azure-pipelines.yml

```diff
@@ -23,3 +23,160 @@ jobs:
+      echo '##vso[task.setvariable variable=AZURE]true'
+    displayName: 'Setting environment variables'
+
+  # Do not require a conda environment
+  - script: |
+      export PATH=$HOME/miniconda3/bin:$PATH
+      ci/code_checks.sh patterns
+    displayName: 'Looking for unwanted patterns'
+    condition: true
+
+  - script: |
+      export PATH=$HOME/miniconda3/bin:$PATH
+      sudo apt-get install -y libc6-dev-i386
+      ci/incremental/install_miniconda.sh
+      ci/incremental/setup_conda_environment.sh
+    displayName: 'Set up environment'
+
+  # Do not require pandas
+  - script: |
+      export PATH=$HOME/miniconda3/bin:$PATH
+      source activate pandas-dev
+      ci/code_checks.sh lint
+    displayName: 'Linting'
+    condition: true
+
+  - script: |
+      export PATH=$HOME/miniconda3/bin:$PATH
+      source activate pandas-dev
+      ci/code_checks.sh dependencies
+    displayName: 'Dependencies consistency'
+    condition: true
+
+  - script: |
+      export PATH=$HOME/miniconda3/bin:$PATH
+      source activate pandas-dev
+      ci/incremental/build.sh
+    displayName: 'Build'
+    condition: true
+
+  # Require pandas
+  - script: |
+      export PATH=$HOME/miniconda3/bin:$PATH
+      source activate pandas-dev
+      ci/code_checks.sh code
+    displayName: 'Checks on imported code'
+    condition: true
+
+  - script: |
+      export PATH=$HOME/miniconda3/bin:$PATH
+      source activate pandas-dev
+      ci/code_checks.sh doctests
+    displayName: 'Running doctests'
+    condition: true
+
+  - script: |
+      export PATH=$HOME/miniconda3/bin:$PATH
+      source activate pandas-dev
+      ci/code_checks.sh docstrings
+    displayName: 'Docstring validation'
+    condition: true
+
+  - script: |
+      export PATH=$HOME/miniconda3/bin:$PATH
+      source activate pandas-dev
+      pytest --capture=no --strict scripts
+    displayName: 'Testing docstring validaton script'
+    condition: true
+
+  - script: |
+      export PATH=$HOME/miniconda3/bin:$PATH
+      source activate pandas-dev
+      doc/make.py html
+    displayName: 'Building docs'
+    condition: true
+
+  - script: |
+      if [ "$(Build.SourceBranch)" == "refs/heads/master" ]; then
+          export NAME="master"
+      else
+          export NAME=$(System.PullRequest.PullRequestId)
+      fi
+      az extension add --name storage-preview
+      az storage blob upload-batch --connection-string $CONNECTION_STRING \
+                                   --source $SOURCE \
+                                   --destination $DESTINATION \
+                                   --destination-path $NAME
+      echo "Documentation uploaded to https://pandas.blob.core.windows.net/docs/$NAME"
+    displayName: 'Publishing docs (Azure storage)'
+    condition: true
+    env:
+      CONNECTION_STRING: $(AZURE_STORAGE_CONNECTION_STRING)
+      SOURCE: $(Build.SourcesDirectory)/doc/build/html/
+      DESTINATION: 'docs'
```

Comment:
I realize we haven't gotten to the bottom of the secrets/connection string thing, but also, the Azure static sites feature [requires that your destination container](https://docs.microsoft.com/azure/storage/blobs/storage-blob-static-website) be the string `$web`. I suspect you can publish a `docs` folder within that container.